UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentCallBackServiceTest.cs

```diff
@@ -0,0 +1,147 @@
+        }
+
+        #region GetCallBackResponse
+        public CallBackResponse GetCallBackResponse()
+        {
+            return new CallBackResponse() 
+            {
+                Specversion="1.0",
+                Type = "test",
+                Source = "test",
+                Subject = "test",
+                Time = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture),
+                DataContentType = "application/json",
+                Data = new ExchangeSetResponse()
+            };
+        }
+        #endregion
+
+        #region GetSalesCatalogueServiceResponse
+        private static SalesCatalogueProductResponse GetSalesCatalogueServiceResponse()
+        {
+            return new SalesCatalogueProductResponse()
+            {
+                ProductCounts = new ProductCounts()
+                {
+                    RequestedProductCount = 12,
+                    RequestedProductsAlreadyUpToDateCount = 5,
+                    RequestedProductsNotReturned = new List<RequestedProductsNotReturned>
+                    {
+                        new RequestedProductsNotReturned()
+                        {
+                            ProductName = "test",
+                            Reason = "notfound"
+                        }
+                    },
+                    ReturnedProductCount = 4
+                },
+                Products = new List<Products> {
+                            new Products {
+                                ProductName = "DE5NOBRK",
+                                EditionNumber = 0,
+                                UpdateNumbers = new List<int?> {0,1},
+                                FileSize = 400
+                            }
+                        }
+            };
+        }
+        #endregion
+
+        #region GetScsResponseQueueMessage
+        private SalesCatalogueServiceResponseQueueMessage GetScsResponseQueueMessage()
+        {
+            return new SalesCatalogueServiceResponseQueueMessage
+            {
+                BatchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272",
+                FileSize = 4000,
+                ScsResponseUri = "https://test/ess-test/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272.json",
+                CallbackUri = "https://test-callbackuri.com",
+                CorrelationId = "727c5230-2c25-4244-9580-13d90004584a"
+            };
+        }
+        #endregion
+
+        [Test]
+        public async Task WhenValidCallBackUriInRequest_ThenSendCallBackReponseReturnsTrue()
+        {
+            string postBodyParam = "This should be replace by actual value when param passed to api call";
+            string uriParam = null;
+            HttpMethod httpMethodParam = null;
+
+            SalesCatalogueProductResponse salesCatalogueProductResponse = GetSalesCatalogueServiceResponse();
+            SalesCatalogueServiceResponseQueueMessage scsResponseQueueMessage = GetScsResponseQueueMessage();
```

Comment:
Kindly correct typo of response in both test case name. Also declare common variables at top and assign in setup method instead of declaring two times in both test cases.

public string postBodyParam;
        public string uriParam;
        public HttpMethod httpMethodParam;
        public SalesCatalogueProductResponse salesCatalogueProductResponse;
        public SalesCatalogueServiceResponseQueueMessage scsResponseQueueMessage;
In setup()
postBodyParam = "This should be replace by actual value when param passed to api call";
            uriParam = null;
            httpMethodParam = null;
            salesCatalogueProductResponse = GetSalesCatalogueServiceResponse();
            scsResponseQueueMessage = GetScsResponseQueueMessage();

Reply:
as per the suggestion i have made required changes in both test cases.